nautobot/core/models/fields.py

```diff
@@ -382,6 +383,24 @@ def formfield(self, **kwargs):
         )
 
 
+class ChoiceArrayField(JSONArrayField):
```

Comment:
Should we just extend JSONArrayField to support choices instead of creating a new field? Django doesn't have any specific choice fields, does it?

Reply:
They do not, and the ArrayField is Postgres only, so we created the Custom JSONArrayFfield.  I created this so that I could add the choices and change the generated form.

Reply:
Could we add this functionality to `JSONArrayField` if the choices are provided on `__init__`?